### DISCUSSION TITLE
Are async computed values possible?

### DISCUSSION THREAD

**Discussion by sitharus** [mobxjs/mobx#2570] [Q&A]
I've been looking at this problem for a while and I suspect I've got the wrong approach, hopefully someone can point me in the right direction.

I have a form that has a list driven from an API based on the value selected in another part of the form. Like a country/state selector, but the second list can't be created statically.

This looks like it's best solved with a computed that observes the first item's value, but a computed can't wait on an async computation, and there's nothing in mobx-utils that seems to match. This makes me think I'm approaching the problem wrong.

**Comment by danielkcz** [comment]
Can you maybe set up a small example with eg. CodeSandbox? I know what you mean, but technically async = Promise. I am not sure though if you return a new Promise from computed every time if it's possible to properly react to its changes. Maybe it would need to utilize `observable.box` to wrap that Promise ... but that's getting ugly.

I suppose, if this would be possible, we would not need to have such [a complex approach for async actions](https://mobx.js.org/actions.html#asynchronous-actions). So my bets are it's not possible.

**Reply by jeremy-coleman** [reply]
https://github.com/danielearwicker/computed-async-mobx

**Reply by sitharus** [reply]
That doesn't work on MobX 6, though I guess I can adapt the concept...

**Reply by sitharus** [reply]
I can make an observable from a promise, so I guess a computed that returns an observable could work. I'll have a play.
